contentful/apps
# feat: display error message when content type or channel is not available [INTEG-1597]

## Purpose

For configured notifications, if the selected content type or channel is no longer available (e.g. it was deleted), then the user should see an error message on that notification.

## Approach

I first updated the app installation parameters so that the content type name would be saved. This allows the name to still display on the notification after the content type is deleted, rather than it showing "Content type not found" as the name.

I updated the existing `ErrorMessage` component to add additional props to make it reusable for this use case as well as the existing one in the selection modals.

I did not any indication of a failing notification for the notification "view" mode, as I wanted to check in with design on that. For now, the error only displays when you open a configured notification. See below for the errors displayed when the content type or channel saved in a notification is not found:

<img width="841" alt="Screenshot 2024-03-29 at 9 22 57 AM" src="https://github.com/contentful/apps/assets/62958907/b0b99a29-3f57-4d48-8d9e-052c32eab302">

## Testing steps

Tested deleting a configured content type and channel to see the new error message.

## Breaking Changes

Because this updates the app installation parameters, the app will likely need to be reinstalled in order for it to work correctly and not throw an error.

## Dependencies and/or References



## Deployment
